src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/RealWorldContextTests.cs

```diff
@@ -882,5 +882,309 @@ public virtual void NullableStruct()
             Assert.Equal("Doe", person.Value.LastName);
         }
+
+        [Fact]
+        public void JsonIgnore_CovariantBaseNotIgnored_CovariantDerivedNotIgnored()
+        {
+            var derived = new CovariantBaseNotIgnored_CovariantDerivedNotIgnored();
+
+            if (DefaultContext.JsonSourceGenerationMode == JsonSourceGenerationMode.Serialization)
+            {
+                Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(derived, DefaultContext.CovariantBaseNotIgnored_CovariantDerivedNotIgnored));
+            }
+            else
+            {
+                string json = JsonSerializer.Serialize(derived, DefaultContext.CovariantBaseNotIgnored_CovariantDerivedNotIgnored);
+                JsonTestHelper.AssertJsonEqual("{\"Id\":\"CovariantBaseNotIgnored_CovariantDerivedNotIgnored\"}", json);
+                derived = JsonSerializer.Deserialize(json, DefaultContext.CovariantBaseNotIgnored_CovariantDerivedNotIgnored);
+                Assert.Equal("CovariantBaseNotIgnored_CovariantDerivedNotIgnored", derived.Id);
+            }
+        }
+
+        [Fact]
+        public void JsonIgnore_CovariantBaseNotIgnored_CovariantDerivedGenericNotIgnoredString()
+        {
+            var derived = new CovariantBaseNotIgnored_CovariantDerivedGenericNotIgnored<string>();
+            if (DefaultContext.JsonSourceGenerationMode == JsonSourceGenerationMode.Serialization)
+            {
+                Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(derived, DefaultContext.CovariantBaseNotIgnored_CovariantDerivedGenericNotIgnoredString));
+            }
+            else
+            {
+                string json = JsonSerializer.Serialize(derived, DefaultContext.CovariantBaseNotIgnored_CovariantDerivedGenericNotIgnoredString);
+                JsonTestHelper.AssertJsonEqual("{\"Id\":\"CovariantBaseNotIgnored_CovariantDerivedGenericNotIgnored\"}", json);
+
+                derived = JsonSerializer.Deserialize(json, DefaultContext.CovariantBaseNotIgnored_CovariantDerivedGenericNotIgnoredString);
+                Assert.Equal("CovariantBaseNotIgnored_CovariantDerivedGenericNotIgnored", derived.Id);
+            }
+        }
+
+        [Fact]
+        public void JsonIgnore_CovariantBaseNotIgnored_CovariantDerivedIgnored()
+        {
+            var derived = new CovariantBaseNotIgnored_CovariantDerivedIgnored();
+
+            if (DefaultContext.JsonSourceGenerationMode == JsonSourceGenerationMode.Serialization)
+            {
+                Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(derived, DefaultContext.CovariantBaseNotIgnored_CovariantDerivedIgnored));
+            }
+            else
+            {
+                string json = JsonSerializer.Serialize(derived, DefaultContext.CovariantBaseNotIgnored_CovariantDerivedIgnored);
+                JsonTestHelper.AssertJsonEqual("{\"Id\":\"CovariantBaseNotIgnored_CovariantDerivedIgnored\"}", json);
```

Comment:
so this effectively means that `[JsonIgnore]` on the derived overriden member is effectively useless? I'd expect either ignore or error here